engine/ipc.go

```diff
@@ -48,6 +48,25 @@ type SeesawEngine struct {
 	engine *Engine
 }
 
+// accessCheck performs an access check based on the given context, vserver
+// and username.
+func (s *SeesawEngine) accessCheck(ctx *ipc.Context, vserver, username string) (string, error) {
```

Comment:
why do you need to pass username while ctx is already holding it?

Reply:
Good point, removed.